app/controllers/api/v3/facilities_controller.rb

```diff
@@ -26,7 +26,7 @@ def disable_audit_logs?
   def transform_to_response(facility)
     Api::V3::FacilityTransformer
       .to_response(facility)
-      .merge(sync_region_id: facility.block_region_id)
+      .merge(sync_region_id: sync_region_id(facility))
```

Comment:
This change is required for two reasons:
- This is the primary fix. The fix in `sync_region_modified?` captures an edge case. We should set the sync region id that we want the client to send back.
- So that the app can figure out which records to purge. If we let this be set to block id, the app _will_ purge records outside the block for MOs. [This](https://app.clubhouse.io/simpledotorg/story/1413/regularly-purge-data-outside-the-current-sync-group) has a good explanation of how the purge works.

Reply:
Ah great point about purges. Yup, looks like we're still in a situation where the server dictates what the client's sync radius should be. I think that's fine for now. I understand the context of the primary fix and edge case better now 😅